hack/chains/end-to-end-demo.sh

```diff
@@ -0,0 +1,138 @@
+fi
+
+{
+  oc project demo 2>/dev/null 1>&2 && echo -n "ℹ️  Using existing 'demo' OpenShift project"
+} || {
+  echo "📂 Creating a new OpenShift project 'demo'
+"
+  oc new-project demo
+}
+echo "Here is the OpenShift console with the project:
+
+👉 $(oc whoami --show-console)/k8s/cluster/projects/demo
+"
+
+echo "♾️  Setting up pipelines
+"
+kubectl apply -k "${BUILD_DEFINITIONS_DIR}/hack/test-build"
+kubectl apply -k "${BUILD_DEFINITIONS_DIR}/pipelines/hacbs"
+
+# This is for the build pipeline
+kubectl create secret docker-registry redhat-appstudio-staginguser-pull-secret --from-file=.dockerconfigjson="${HOME}/.docker/config.json" --dry-run=client -o yaml | kubectl apply -f -
+
+# This is for the chains-controller
+kubectl create secret docker-registry quay-pull-secret --from-file=.dockerconfigjson="${HOME}/.docker/config.json" -n tekton-chains --dry-run=client -o yaml | kubectl apply -f -
+kubectl patch sa pipeline -n tekton-chains -p '{"imagePullSecrets": [{"name": "quay-pull-secret"}]}'
+
+echo "
+🏃 Running a build pipeline
+
+💲 build-definitions/hack/test-build.sh ${APP_GIT_REPOSITORY} ${APP_BUILDER}
+"
+# see https://serverfault.com/a/989827
+{ BUILD_PIPELINE_RUN=$("${BUILD_DEFINITIONS_DIR}/hack/test-build.sh" "${APP_GIT_REPOSITORY}" "${APP_BUILDER}" | tee /dev/fd/3 | grep 'PipelineRun started: ' | sed -e 's/PipelineRun started: //'); } 3<&1
+tkn pipelinerun logs --follow "${BUILD_PIPELINE_RUN}"
+BUILD_OUTPUT_IMAGE_REF=$(tkn pipelinerun describe "${BUILD_PIPELINE_RUN}" -o jsonpath='{.spec.params[?(@.name=="output-image")].value}')
+
+echo "
+ℹ️  Build pipeline finished
+
+Make note that some of the steps could have been skipped over because of the
+pipeline caching.
+
+Here is the pipeline visualisation:
+
+👉 $(oc whoami --show-console)/k8s/ns/demo/tekton.dev~v1beta1~PipelineRun/${BUILD_PIPELINE_RUN}
+"
+
+RELEASE_OUTPUT_IMAGE_REF="${BUILD_OUTPUT_IMAGE_REF%:*}:production"
+
+echo "
+🏃 Running a sample release pipeline (to demo Enterprise Contract task)
+
+💲 infra-deployments/hack/chains/release-pipeline-with-ec-demo.sh" "${BUILD_OUTPUT_IMAGE_REF}" "${RELEASE_OUTPUT_IMAGE_REF}
+"
+kubectl create secret docker-registry release-demo --from-file=.dockerconfigjson="${HOME}/.docker/config.json" --dry-run=client -o yaml | kubectl apply -f -
+oc secrets link pipeline release-demo --for=pull,mount
+
+# Enterprise Contract policy configuration
+kubectl create configmap ec-policy --from-file=policy.json=<(echo '{"non_blocking_checks":["not_useful","test:conftest-clair"]}'|jq .) 2>/dev/null || echo "ℹ️  Using existing Enterprise Contract policy configuration in ec-policy ConfigMap"
+
+"${HACK_CHAINS_DIR}/copy-public-sig-key.sh"
+TASK_BUNDLE=quay.io/redhat-appstudio/appstudio-tasks:$(git ls-remote --heads https://github.com/redhat-appstudio/build-definitions.git refs/heads/main|cut -f 1)-2
+export TASK_BUNDLE
+# For simplicity tag with `production`
```

Comment:
Comment belongs above line 121 I think.

Reply:
Yup stuff got moved around and I forgot to move the comment with it